packages/engine/src/camera/systems/CameraSystem.ts

```diff
@@ -267,12 +273,35 @@ function updateSpectator(cameraEntity: Entity) {
   const spectator = getComponent(cameraEntity, SpectatorComponent)
 
   const networkCameraEntity = world.getOwnedNetworkObjectWithComponent(spectator.userId, NetworkCameraComponent)
+
+  if (!networkCameraEntity) {
+    // Go back to normal mode
+    removeComponent(cameraEntity, SpectatorComponent)
```

Comment:
We should stay in spectator mode until the user presses an "exit spectator" button